### PR TITLE
Os 830/fix open data

### PR DESCRIPTION
**Pull request by flying-pi**

### CODE REVIEW

File: ecommerce/extensions/basket/views.py [hunk 1]
```diff
@@ -379,14 +381,16 @@ def get_context_data(self, **kwargs):
         except ValueError:
             total_benefit = None
 
+
```
Comment (idegtiarov) [inline ecommerce/extensions/basket/views.py]: Blank line is not need could be removed.

File: ecommerce/extensions/basket/views.py [hunk 2]
```diff
@@ -379,14 +381,16 @@ def get_context_data(self, **kwargs):
...
         context.update({
             'formset_lines_data': zip(formset, lines_data),
             'free_basket': context['order_total'].incl_tax == 0,
             'homepage_url': get_lms_url(''),
             'min_seat_quantity': 1,
             'payment_processors': payment_processors,
             'total_benefit': total_benefit,
-            'is_pay_for_credit': self.request.GET.get('is_pay_for_credit', True),
+            'is_pay_for_credit': get_credit_payment_info(self.request).get('is_pay_for_credit', True),
+            'credit_payment_info': self.request.GET.get('credit_payment_info', True),
```
Comment (idegtiarov) [inline ecommerce/extensions/basket/views.py]: Clarify default parameter

File: ecommerce/extensions/checkout/views.py [hunk 3]
```diff
@@ -214,6 +215,9 @@ def get_order_verification_context(self, order):
 
 class EnrollToCreditAndShowDashboard(View):
     def get(self, request):
+        payment_info = get_credit_payment_info(request)
+        if payment_info.get('user','') != request.user.username or payment_info.get('is_pay_for_credit', True):
```
Comment (idegtiarov) [inline ecommerce/extensions/checkout/views.py]: default values are not need here

File: ecommerce/extensions/checkout/views.py [hunk 4]
```diff
@@ -214,6 +215,9 @@ def get_order_verification_context(self, order):
 
 class EnrollToCreditAndShowDashboard(View):
```
Comment (idegtiarov) [inline ecommerce/extensions/checkout/views.py:216]: Please add comments here

File: ecommerce/ecommece_util.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+from django.conf import settings
+
+from Crypto.Cipher import AES
+
+
+def get_credit_payment_info(request):
+    decryptor = AES.new(key=hashlib.md5(settings.EDX_API_KEY).hexdigest(), mode=AES.MODE_ECB)
+    credit_payment_info_from_request = request.GET.get('credit_payment_info', '')
+    credit_payment_info = b64decode(credit_payment_info_from_request.replace(' ', '+'))
```
Comment (idegtiarov) [inline ecommerce/ecommece_util.py]: Please add comment here

**Review by idegtiarov — CHANGES_REQUESTED** [review]

File: ecommerce/extensions/basket/views.py (hunk 1 above)
Comment (flying-pi) [reply ecommerce/extensions/basket/views.py]: done

File: ecommerce/extensions/basket/views.py (hunk 2 above)
Comment (flying-pi) [reply ecommerce/extensions/basket/views.py]: done

File: ecommerce/extensions/checkout/views.py (hunk 3 above)
Comment (flying-pi) [reply ecommerce/extensions/checkout/views.py]: done

File: ecommerce/extensions/checkout/views.py (hunk 4 above)
Comment (flying-pi) [reply ecommerce/extensions/checkout/views.py:216]: done

File: ecommerce/ecommece_util.py (hunk 5 above)
Comment (flying-pi) [reply ecommerce/ecommece_util.py]: done

**Review by idegtiarov — APPROVED** [review]

**Pull request merged**